internal/cmd/ske/kubeconfig/login/login.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		return nil, fmt.Errorf("ConvertToVersion: %w", err)
+	}
+
+	execCredential, ok := obj.(*clientauthenticationv1.ExecCredential)
+	if !ok {
+		return nil, fmt.Errorf("conversion to ExecCredential failed")
+	}
+	if execCredential == nil || execCredential.Spec.Cluster == nil {
+		return nil, fmt.Errorf("ExecCredential contains not all needed fields")
+	}
+	config := &SKEClusterConfig{}
+	err = json.Unmarshal(execCredential.Spec.Cluster.Config.Raw, config)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal: %w", err)
+	}
+
+	return &inputModel{
+		ClusterName: config.ClusterName,
+		ProjectId:   config.STACKITProjectID,
+		CacheKey:    fmt.Sprintf("ske-login-%x", sha256.Sum256([]byte(execCredential.Spec.Cluster.Server))),
+	}, nil
+}
+
+func getCachedKubeConfig(key string) *rest.Config {
+	cachedKubeconfig, err := cache.GetObject(key)
+	if err != nil {
+		return nil
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(cachedKubeconfig)
+	if err != nil {
+		return nil
+	}
+
+	return restConfig
+}
+
+func GetAndOutputKubeconfig(ctx context.Context, cmd *cobra.Command, apiClient *ske.APIClient, model *inputModel, fallbackToCache bool, cachedKubeconfig *rest.Config) error {
+	req := buildRequest(ctx, apiClient, model)
+	kubeconfigResponse, err := req.Execute()
+	if err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("request kubeconfig: %w", err)
+	}
+
+	kubeconfig, err := clientcmd.RESTConfigFromKubeConfig([]byte(*kubeconfigResponse.Kubeconfig))
+	if err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("parse kubeconfig: %w", err)
+	}
+	if err = cache.PutObject(model.CacheKey, []byte(*kubeconfigResponse.Kubeconfig)); err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("cache kubeconfig: %w", err)
+	}
+
+	return output(cmd, model.CacheKey, kubeconfig)
+}
+
+func buildRequest(ctx context.Context, apiClient *ske.APIClient, model *inputModel) ske.ApiCreateKubeconfigRequest {
+	req := apiClient.CreateKubeconfig(ctx, model.ProjectId, model.ClusterName)
+	expirationSeconds := strconv.Itoa(expirationSeconds)
+
+	return req.CreateKubeconfigPayload(ske.CreateKubeconfigPayload{ExpirationSeconds: &expirationSeconds})
+}
+
+func output(cmd *cobra.Command, cacheKey string, kubeconfig *rest.Config) error {
+	outputExecCredential, err := parseKubeConfigToExecCredential(kubeconfig)
+	if err != nil {
+		_ = cache.DeleteObject(cacheKey)
+		return fmt.Errorf("convert to ExecCredential: %w", err)
+	}
+
+	output, err := json.Marshal(outputExecCredential)
+	if err != nil {
+		_ = cache.DeleteObject(cacheKey)
+		return fmt.Errorf("marshal ExecCredential: %w", err)
+	}
+
+	cmd.Print(string(output))
+	return nil
+}
+
+func parseKubeConfigToExecCredential(kubeconfig *rest.Config) (*clientauthenticationv1.ExecCredential, error) {
```

Comment:
consider adding unit test for this function

Reply:
What should I test there? The function basically only puts part of the kubeconfig into the expected struct.

Reply:
You can test that the returned struct is like expected